app/models/katello/concerns/subscription_facet_host_extensions.rb

```diff
@@ -45,32 +49,49 @@ module ClassMethods
         def find_by_activation_key(_key, operator, value)
           conditions = sanitize_sql_for_conditions(["#{Katello::ActivationKey.table_name}.name #{operator} ?", value_to_sql(operator, value)])
           hosts = ::Host::Managed.joins(:activation_keys).where(conditions)
-          if hosts.empty?
-            { :conditions => "1=0" }
-          else
-            { :conditions => "#{::Host::Managed.table_name}.id IN (#{hosts.pluck(:id).join(',')})" }
-          end
+          return_hosts(hosts)
         end
 
         def find_by_activation_key_id(_key, operator, value)
           conditions = sanitize_sql_for_conditions(["#{Katello::ActivationKey.table_name}.id #{operator} ?", value_to_sql(operator, value)])
           hosts = ::Host::Managed.joins(:activation_keys).where(conditions)
-          if hosts.empty?
-            { :conditions => "1=0" }
-          else
-            { :conditions => "#{::Host::Managed.table_name}.id IN (#{hosts.pluck(:id).join(',')})" }
-          end
+          return_hosts(hosts)
         end
 
         def find_by_hypervisor_host(_key, operator, value)
           conditions = sanitize_sql_for_conditions(["#{::Host.table_name}.name #{operator} ?", value_to_sql(operator, value)])
           hosts = ::Host.where(conditions)
           hosts = ::Host.joins(:subscription_facet).where("#{Katello::Host::SubscriptionFacet.table_name}.hypervisor_host_id" => hosts)
+          return_hosts(hosts)
+        end
 
+        def find_by_subscription_name(_key, operator, value)
+          conditions = sanitize_sql_for_conditions(["#{Katello::Subscription.table_name}.name #{operator} ?", value_to_sql(operator, value)])
+          sub_facets = ::Katello::Host::SubscriptionFacet.joins(pools: :subscription).where(conditions)
+          host_ids = sub_facets.select(:host_id)
+          return_hosts_by_id(host_ids)
+        end
+
+        def find_by_subscription_id(_key, operator, value)
+          conditions = sanitize_sql_for_conditions(["#{Katello::Subscription.table_name}.id #{operator} ?", value_to_sql(operator, value)])
+          sub_facets = ::Katello::Host::SubscriptionFacet.joins(pools: :subscription).where(conditions)
+          host_ids = sub_facets.select(:host_id)
+          return_hosts_by_id(host_ids)
+        end
```

Comment:
So this it a bit confusing, but to the user 'subscriptions' are actually represented as 'pools' internally.  Hitting the subscriptions api returns a pool with subscription info tacked on.  

So search for 'subscription_id = 3' really needs to search for the internal pool id instead of joining to the subscriptions table and querying based on that.